sql/core/src/main/scala/org/apache/spark/sql/execution/SparkQl.scala

```diff
@@ -16,18 +16,38 @@
  */
 package org.apache.spark.sql.execution
 
+import scala.collection.mutable.ArrayBuffer
+
 import org.apache.spark.sql.{AnalysisException, SaveMode}
-import org.apache.spark.sql.catalyst.{CatalystQl, TableIdentifier}
+import org.apache.spark.sql.catalyst.TableIdentifier
 import org.apache.spark.sql.catalyst.analysis.UnresolvedRelation
-import org.apache.spark.sql.catalyst.parser.{ASTNode, ParserConf, SimpleParserConf}
+import org.apache.spark.sql.catalyst.expressions.{Ascending, Descending}
+import org.apache.spark.sql.catalyst.parser.{ASTNode, CatalystQl, ParserConf, ParserSupport, SimpleParserConf}
 import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, OneRowRelation}
 import org.apache.spark.sql.catalyst.plans.logical
+import org.apache.spark.sql.execution.commands._
 import org.apache.spark.sql.execution.datasources._
 import org.apache.spark.sql.types.StructType
 
 private[sql] class SparkQl(conf: ParserConf = SimpleParserConf()) extends CatalystQl(conf) {
+  import ParserSupport._
+
   /** Check if a command should not be explained. */
-  protected def isNoExplainCommand(command: String): Boolean = "TOK_DESCTABLE" == command
+  protected def isNoExplainCommand(command: String): Boolean =
+    "TOK_DESCTABLE" == command || "TOK_ALTERTABLE" == command
+
+  protected def extractProps(
+      node: ASTNode,
+      firstLevelNodeStr: String,
+      secondLevelNodeStr: String): Seq[(String, String)] = node match {
+    case Token(firstLevelNodeStr, options) =>
```

Comment:
e.g.
```
scala> object Token { def unapply(x: String): Some[(String, String)] = Some((x, x)) }
scala> val matchString = "abc"
scala> "def" match {
     |   case Token(matchString, _) => true
     |   case _ => false
     | }
res8: Boolean = true
```
It should have returned false because `def` doesn't match `abc`